docs/concepts/what-is-a-workflow-execution.md

```diff
@@ -2,57 +2,122 @@
+A Workflow Execution Chain is a sequence of Workflow Executions that share the same Workflow Id.
+Each link in the Chain is often called a Workflow Run.
+Each Workflow Run in the sequence is connected by one of the following:
 
-The "Continue-as-new" feature completes the current Workflow execution and automatically starts a new execution with the same Workflow Id, but different run Id, passing it the appropriate parameters for it to continue.
-This keeps the event history within limits, but continues the logic execution.
+- [Temporal Cron Job](/docs/concepts/what-is-a-temporal-cron-job)
+- [Continue As New](/docs/concepts/what-is-continue-as-new)
+* [Retries](/docs/concepts/what-is-a-retry-policy)
 
-**How can I load test Workflow Executions?**
+A Workflow Execution is uniquely identified by its [Namespace](/docs/concepts/what-is-a-namespace), [Workflow Id](/docs/concepts/what-is-a-workflow-id), and [Run Id](/docs/concepts/what-is-a-run-id).
 
-The [Temporal stress testing blog post](https://docs.temporal.io/blog/temporal-deep-dive-stress-testing) covers many different scenarios under which we test Workflow Executions.
+The [Workflow Execution Timeout](/docs/concepts/what-is-a-workflow-execution-timeout) applies to a Workflow Execution Chain.
+The [Workflow Run Timeout](/docs/concepts/what-is-a-workflow-run-timeout) applies to a single Workflow Execution (Workflow Run).
 
-**Implementation guides:**
+### Event loop
 
-- [How to spawn a Workflow Execution in Go](/docs/go/how-to-spawn-a-workflow-execution-in-go)
+A Workflow Execution is made up of a set of [Events](/docs/concepts/what-is-an-event) called an [Event History](/docs/concepts/what-is-an-event-history).
+Events are created by the Temporal Cluster in response to either Commands or actions requested by a Temporal Client (such as a request to spawn a Workflow Execution).
+
+![Workflow Execution](/diagrams/workflow-execution-swim-lane-01.svg)
+
+### Time constraints
+
+**Is there a limit to how long Workflows can run?**
+
+No, there is no time constraint on how long a Workflow Execution can be Running.
+
+However, Workflow Executions intended to run indefinitely should be written with some care.
+The Temporal Cluster stores the complete Event History for the entire lifecycle of a Workflow Execution.
+There is a hard limit of 50,000 Events in a Workflow Execution Event History, as well as a hard limit of 50MB in terms of size.
+The Temporal Cluster logs a warning at every 10,000 Events.
+
+To prevent "runaway" Workflow Executions you can use the Workflow Execution Timeout and/or the Workflow Run Timeout.
+A Workflow Execution Timeout can be used to limit the duration of Workflow Execution Chain, and a Workflow Run Timeout can be used to limit the duration an individual Workflow Execution (Run).
+
+You can use the [Continue-As-New](/docs/concepts/what-is-continue-as-new) feature to create a new Workflow Execution.
```

Comment:
Should we mention that closing previous execution and starting the new one is an atomic operation when continue as new is used?